providers/salesforce/bulk-info.go

```diff
@@ -0,0 +1,129 @@
+package salesforce
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/amp-labs/connectors/common"
+)
+
+// GetBulkQueryInfo returns information status about a Query Job,
+// which was created via LaunchBulkRead or LaunchBulkQuery.
+// https://developer.salesforce.com/docs/atlas.en-us.api_asynch.meta/api_asynch/query_get_one_job.htm
+func (c *Connector) GetBulkQueryInfo(ctx context.Context, jobId string) (*GetJobInfoResult, error) {
+	location, err := c.getRestApiURL("jobs/query", jobId)
+	if err != nil {
+		return nil, err
+	}
+
+	res, err := c.Client.Get(ctx, location.String())
+	if err != nil {
+		return nil, fmt.Errorf(
+			"failed to get bulk query info for job '%s': %w",
+			jobId,
+			errors.Join(err, common.ErrRequestFailed),
+		)
+	}
+
+	return common.UnmarshalJSON[GetJobInfoResult](res)
+}
+
+// GetBulkQueryResults returns completed data from bulk query.
+// https://developer.salesforce.com/docs/atlas.en-us.api_asynch.meta/api_asynch/query_get_job_results.htm
+func (c *Connector) GetBulkQueryResults(ctx context.Context, jobId string) (*http.Response, error) {
+	location, err := c.getRestApiURL("jobs/query/", jobId, "/results")
+	if err != nil {
+		return nil, err
+	}
+
+	req, err := common.MakeJSONGetRequest(ctx, location.String(), []common.Header{
+		{
+			Key:   "Accept",
+			Value: "text/csv",
```

Comment:
From the docs, seems like we can also get JSON results, can we allow for that? That'll be useful for reads.

Reply:
Sadly, no. I was wondering that too. According to the documentation it says the only supported option is CSV. The option is there but with one choice ahaha, is it even an option, never understood such motive.
CreateJob: https://developer.salesforce.com/docs/atlas.en-us.api_asynch.meta/api_asynch/query_create_job.htm
![image](https://github.com/user-attachments/assets/ec9f1d9c-e6c3-4941-b996-c1b42e10c6d0)